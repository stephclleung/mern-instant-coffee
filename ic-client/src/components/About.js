import React from 'react';
import { connect } from 'react-redux';
import { setLocation } from '../actions/location';
import { Container, Row, Col } from 'reactstrap';

export const About = (props) => {
    props.setLocation();
    return (
        <Container>
            <Row>
                <Col xs="auto">
                    <img src={"https://i.imgur.com/ebLwdYJ.png"} alt={"this is steph."} style={{ width: 100 + "%" }}></img>
                </Col>
            </Row>
            <Row>
                <Col xs="auto" >
                    <h3>Hi there!
                    Thanks for checking out my app.</h3>
                    I made this app as a practice for developing with MongoDB, Express, React and Node.
                    <p>Feel free to poke around.</p>
                    <p>If you haven't already done so, please also visit my portfolio at</p>
                    <div className="externalLinks"><center><a className="links" href="https://www.cl-leung.com" target="_blank" rel="noopener noreferrer"> www.cl-leung.com.</a></center></div>
                    <p>Hope you have a great day :).</p>
                </Col>
            </Row>
        </Container>
    )

}

const mapDispatchToProps = (dispatch) => {
    return {
        setLocation: () => dispatch(setLocation('about'))
    }
}

export default connect(undefined, mapDispatchToProps)(About);
//export default About;