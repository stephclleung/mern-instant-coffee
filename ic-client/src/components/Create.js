import React from 'react';
import { Link } from 'react-router-dom';
import InstantCoffeeForm from './InstantCoffeeForm';

//TODO : dispatch "add" when InstantCoffeeForm submits.
class Create extends React.Component {
   
    
    render() {
        return (
            <div>
                <InstantCoffeeForm />
                <Link to="/">Back</Link>
            </div>
        );
    }
};

export default Create;
