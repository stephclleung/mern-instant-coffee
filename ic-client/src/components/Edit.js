import React from 'react';
import InstantCoffeeForm from './InstantCoffeeForm';
import { connect } from 'react-redux';
import { removeInstantCoffeeFromDB, editInstantCoffeeToDB } from '../actions/instant-coffee';
import { handleOtherError } from '../actions/error';
import { setLocation } from '../actions/location';
import { Button } from 'reactstrap';


export class Edit extends React.Component {
    onRemoveCoffee = () => {
        this.props.removeInstantCoffeeFromDB(this.props.instantCoffee.id)
        this.props.history.push('/')
    }
    onEditCoffee = (coffeeUpdates) => {
        if (Object.entries(coffeeUpdates.changedFields).length !== 0) {
            try {
                this.props.editInstantCoffeeToDB(this.props.instantCoffee.id, coffeeUpdates);
            } catch (error) {
                this.props.handleOtherError();
                this.props.history.push('/')
            }
        }
        this.props.history.push('/')
    }
    render() {
        this.props.setLocation();
        return (
            <div>
                <Button onClick={this.onRemoveCoffee} className="col-lg-6 mb-3 btn-block align-item-middle">Remove Coffee</Button>
                <InstantCoffeeForm instantCoffee={this.props.instantCoffee} onSubmit={this.onEditCoffee} />
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        removeInstantCoffeeFromDB: (id) => dispatch(removeInstantCoffeeFromDB(id)),
        editInstantCoffeeToDB: (id, coffeeUpdates) => dispatch(editInstantCoffeeToDB(id, coffeeUpdates)),
        handleOtherError: () => dispatch(handleOtherError()),
        setLocation: () => dispatch(setLocation('edit'))
    }
}

const mapStateToProps = (state, props) => ({
    instantCoffee: state.instantCoffees.find((ic) => ic.id === props.match.params.id)
});

export default connect(mapStateToProps, mapDispatchToProps)(Edit);
//export default Edit;