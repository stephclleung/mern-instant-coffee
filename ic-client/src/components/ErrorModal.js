import React from 'react';
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
import { connect } from 'react-redux';
import { removeError } from '../actions/error'

class ErrorModal extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            toggle: true
        }
    }
    onToggle = () => {
        //set toggle back false
        this.setState((prevState) => ({ toggle: !prevState }))
        //dispatch request to reset state to no error.
        this.props.removeError();
    }
    render() {
        return (
            <Modal
                isOpen={this.state.toggle}
                centered={true}
            >
                <ModalHeader
                    toggle={this.onToggle}
                >
                    Something went wrong.
                </ModalHeader>
                <ModalBody>
                    {this.props.errorMessage}
                </ModalBody>
                <ModalFooter>
                    <Button
                        onClick={this.onToggle}
                    >
                        Ok
                    </Button>
                </ModalFooter>
            </Modal>
        );
    }
};

const mapDispatchToProps = (dispatch) => {
    return {
        removeError: () => dispatch(removeError())
    }
}

export default connect(null, mapDispatchToProps)(ErrorModal);
// export default ErrorModal;