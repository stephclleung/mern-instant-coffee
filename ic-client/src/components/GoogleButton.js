
import { GoogleLogout, GoogleLogin } from 'react-google-login';
import React from 'react';

const GoogleButton = (props) => (
    <div>
        {console.log("Am butt", props)}
        {!!props.user.userName ? (
            <div>
                <p>Name: {props.user.userName}</p>
                <Logout onLogoutSuccess={props.onGoogleLogout} />
            </div>
        ) : (
                <Login onSuccess={props.onGoogleLogin} />
            )}
    </div>
);

const Login = (props) => (
    <div>
        {console.log("Am log in , ", props)}
        <GoogleLogin
            className="002"
            clientId={process.env.REACT_APP_GOOGLE_CLIENT_ID}
            buttonText="Login"
            onRequest={loading}
            onSuccess={props.onSuccess}
            onFailure={tempFunction}
            cookiePolicy={'single_host_origin'}
        />
    </div>
)

const Logout = (props) => (
    <div>
        {console.log("Am log out, ", props)}
        {console.log("Am client ID", process.env.REACT_APP_GOOGLE_CLIENT_ID)}
        <GoogleLogout
            className="001"
            clientId={process.env.REACT_APP_GOOGLE_CLIENT_ID}
            buttonText="Logout"
            onRequest={loading}
            onLogoutSuccess={props.onLogoutSuccess}
            onLogoutFailure={tempFunction}
        >
        </GoogleLogout>
    </div>
)

const tempFunction = (data) => {
    console.log("Temp function fired", data);
}

const loading = () => {
    console.log('loading') // eslint-disable-line
}
export default GoogleButton;