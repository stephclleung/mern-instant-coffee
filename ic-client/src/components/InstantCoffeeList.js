import React from 'react';
import InstantCoffee from './InstantCoffee';
import { connect } from 'react-redux';

// Need all instant coffee, thus need access to state's storage of the coffee list.
export const InstantCoffeeList = (props) => {
    return (
        <div>
            {listAllInstantCoffees(props.instantCoffees)}
        </div>
    );
}

const listAllInstantCoffees = (coffees) => {
    return coffees.map((coffee) => {
        console.log("In list", coffee)
        return <InstantCoffee key={coffee._id} {...coffee} />
    })
}

const mapStateToProps = (state) => {
    return {
        instantCoffees: state.instantCoffees
    }
}

//export default InstantCoffeeList;
export default connect(mapStateToProps)(InstantCoffeeList)