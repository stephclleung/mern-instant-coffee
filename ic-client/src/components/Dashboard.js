import React from 'react';
import InstantCoffeeList from './InstantCoffeeList';

const Dashboard = () =>{
    return (
        <div>
            {console.log("ICL: ", InstantCoffeeList)}
            <p>Dashboard</p>
            <InstantCoffeeList />
        </div>
    );
};

export default Dashboard;
