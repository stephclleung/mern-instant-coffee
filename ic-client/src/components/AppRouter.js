import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import ErrorModal from './ErrorModal';
import Create from './Create';
import Dashboard from './Dashboard';
import NoMatch from './NoMatch';
import Header from './Header';
import Edit from './Edit';
import { Container } from 'reactstrap';
import { connect } from 'react-redux';

const AppRouter = (props) => {
    return (
        <BrowserRouter>
            <Header />
            <Container pt="4">
                <div className="m-3">
                    <Switch>
                        <Route exact={true} path="/" component={Dashboard} />
                        <Route path="/create" component={Create} />
                        <Route path="/login" />
                        <Route path="/edit/:id" component={Edit} />
                        <Route component={NoMatch} />
                    </Switch>
                </div>
            </Container>
            {props.errors.showErrorMessage &&
                <ErrorModal errorMessage={props.errors.errorMessage} showErrorMessage={props.errors.showErrorMessage} />}
        </BrowserRouter>
    );
};

const mapStateToProps = (state) => ({
    errors: state.errors
})

//export default AppRouter;
export default connect(mapStateToProps)(AppRouter);
