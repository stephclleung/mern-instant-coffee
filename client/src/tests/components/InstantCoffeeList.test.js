import React from 'react';
import { InstantCoffeeList } from '../../components/InstantCoffeeList';
import { instantCoffee } from '../fixtures/instant-coffee-data';
import { shallow } from 'enzyme';

test("Should render instant coffee list component correctly", () => {
    const wrapper = shallow(<InstantCoffeeList instantCoffees={instantCoffee} />)
    expect(wrapper).toMatchSnapshot();
});

