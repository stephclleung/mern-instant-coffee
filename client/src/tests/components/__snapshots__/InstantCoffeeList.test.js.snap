// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render instant coffee list component correctly 1`] = `ShallowWrapper {}`;
